Première implémentation du défi 1 - par aquamen

qaco opened the pull request:
Petit bug : dans les formules, il ne faut pas d'espaces entre les paramètres, donc il faut taper =#(1,2,3,4,5) !
Pour compiler : installer cargo et lancer make.
Pour exécuter : exécuter make run.

KrohomKrohom committed:
branch triplek

yurug committed:
Notes.

yurug committed:
Notes.

n3f4s committed:
Init dev directory

Basic makefile calling cargo
Directory architecture
Basic Cargo.toml

JuRolland committed:
Data types

Commit:
correction des types

Commit:
Parser de valeurs simples

Commit:
Moins verbeux

Commit:
Beta du parser

Commit:
premier test unitaire sur le parser !

Commit:
tests unitaires du parser

Commit:
correction du dernier bug

Commit:
use inutile

n3f4s committed:
Move parser to "thread-local" mod

Add some import in the code
Define type Matrix

n3f4s committed:
Add definition of schedule function

n3f4s committed:
Add eval function definition

n3f4s committed:
Define Matrix struct

n3f4s committed:
Add helper trait for types

n3f4s committed:
Implement eval function

n3f4s committed:
Add process function call parse,eval,print

n3f4s committed:
Implement pass-throught sched function

n3f4s committed:
Add printer

n3f4s committed:
Add comments for future improvements

yurug committed:
Prestations de la semaine 3.

yurug committed:
Prestations de la semaine 3.

n3f4s committed:
Implement main

n3f4s committed:
Clean code and fix split file

yurug committed:
Apparition d'un nouveau critère.

yurug committed:
Apparition d'un nouveau critère.

KhawlaMhb commented on defis/1/aquamen/src/process/parser.rs:
```diff
@@ -0,0 +1,159 @@
+use combine::attempt;
```
Hamza évalue modèle de données
Khawla évalue parser
Zhen évalue Evaluateur

Correction:
gère les nombres et les formules bien formées, c'est ce qui est demandé dans la spec 

Complexité:
n*n (nombre de lignes * taille de ligne)? Parcours de la matrice des cellules

Simplicité:
Architecture claire comprise rapidement
Un ficier a un rôle

Lisibilité:
Manque de commentaires sur les spécificités du langage. En plus le langage n'est pas verbeux et utilisation de noms flous (par exemple une variable res) donc du mal à comprendre ce que chaque fonction fait.
A part ça le code parait clair concis et indenté, très lisible

Performance:
I dont't know 

Robustesse:
Gestion des espaces/fichiers mal formés pas présente

Portabilité:
Makefile -> portable

Modularité: 
Un main qui charge les fichiers et appelle le package process ou tout se passe (pas très modulaire)
Parser et evaluation paraissent être des modules différents, or ils font partie d'un même package
Pour le parser: plusieurs petites méthodes courtes gérant un cas chacune, modulaire

Généralité:

Extensibilité:

vanHamza commented on defis/1/aquamen/src/data.rs:
```diff
@@ -0,0 +1,47 @@
+
```
Correction : dans le main on peut dire que le programme il lit le fichier seulement sans donner les fichiers en sortie
Complexité : 
Simplicité: les structures sont simples et repondent a la specification 
Performance : matrix (vec <vec>) pour un grand fichier c est trop long a charger 
Robustesse : 
Portabilité : Rust est portable
Lisibilité : les structures est bien lisible et claire
Modularité: c'est mieux de séparer les données du main en ajoutant un package pour les données
Généralité: on peut bien change le type de la matrix sans touché le code 
Extensibilité: on a la possibilité de rajouter d'autres fonctionnalités au programme
Évolutivité:

dev590t reviewed (commented):
eval.rs est bien

n3f4s committed:
Fix index error

Commit:
Journal du 28/01

Commit:
Journal du 28/01

Commit:
Merge branch 'aquamen/master' of github.com:yurug/pcomp-2019 into aquamen/master

n3f4s committed:
Fix warnings

n3f4s committed:
Fix warnings

n3f4s committed:
Replace Num by Data in eval

Allow for better error handling when printing the result

n3f4s committed:
Replace Num by Data in eval

Allow for better error handling when printing the result

n3f4s committed:
Reflect change of result matrix type

n3f4s committed:
Reflect change of result matrix type

n3f4s committed:
-

n3f4s committed:
-

n3f4s committed:
Add flag for building for release

n3f4s committed:
Add flag for building for release

n3f4s committed:
Add first try of dockerfile

n3f4s committed:
Add first try of dockerfile

yurug committed:
Amine a présenté.

yurug committed:
Amine a présenté.

Commit:
Robustesse sur les espaces

Commit:
Robustesse sur les espaces

JuRolland committed:
New test

JuRolland committed:
New test

JuRolland committed:
Reorganisation + changes.txt computation

JuRolland committed:
Reorganisation + changes.txt computation

JuRolland committed:
Reorganisation + changes.txt computation

JuRolland committed:
Reorganisation + changes.txt computation

JuRolland committed:
Merge branch 'aquamen/master' of github.com:yurug/pcomp-2019 into aquamen/master

Commit:
Encore plus robuste aux espaces

Commit:
Encore plus robuste aux espaces

Commit:
parse changes

Commit:
parse changes

n3f4s committed:
Improve makefile

n3f4s committed:
Improve makefile

n3f4s committed:
Improved Makefile

n3f4s committed:
Improved Makefile

n3f4s committed:
Add helper for running the program

n3f4s committed:
Add helper for running the program

n3f4s committed:
Try to makes run.sh work CL arg from docker run

n3f4s committed:
Try to makes run.sh work CL arg from docker run

n3f4s committed:
-

n3f4s committed:
-

n3f4s committed:
-

n3f4s committed:
-

n3f4s committed:
-

n3f4s committed:
-

n3f4s committed:
-

n3f4s committed:
-

n3f4s committed:
Fix dockerfile

n3f4s committed:
Fix dockerfile

n3f4s committed:
Improve Makefile

n3f4s committed:
Improve Makefile

n3f4s committed:
Make run.sh actually run the program

n3f4s committed:
Make run.sh actually run the program

n3f4s committed:
Fix bug with 0-based indices in formula

n3f4s committed:
Fix bug with 0-based indices in formula

n3f4s committed:
Add CL args for make run

n3f4s committed:
Add CL args for make run

n3f4s committed:
Add make perf for benchs

n3f4s committed:
Add make perf for benchs

n3f4s committed:
Eval returns a list of requirements

Eval now returns a list of cell whose evaluation is required to finish
the computations. Right now those requirements are not used so we use a
singleton type to avoid passing around vectors. The requirements will be
used when we need to split spreadsheets (e.g. for big files or for
parralelism)

n3f4s committed:
Eval returns a list of requirements

Eval now returns a list of cell whose evaluation is required to finish
the computations. Right now those requirements are not used so we use a
singleton type to avoid passing around vectors. The requirements will be
used when we need to split spreadsheets (e.g. for big files or for
parralelism)

n3f4s committed:
Use the full type in pattern matching

n3f4s committed:
Use the full type in pattern matching

n3f4s committed:
Explicit arguments naming

n3f4s committed:
Explicit arguments naming

n3f4s committed:
Add missing ws.sh file

n3f4s committed:
Add process function handling one thread

This function takes care of everything that is happening in a thread. It
will be called by the scheduler and should be able to send and receive
messages from and to the scheduler. Each thread will have to start only
this function.

n3f4s committed:
Add process function handling one thread

This function takes care of everything that is happening in a thread. It
will be called by the scheduler and should be able to send and receive
messages from and to the scheduler. Each thread will have to start only
this function.

n3f4s committed:
Remove code from sched that should be in process

n3f4s committed:
Remove code from sched that should be in process

JuRolland committed:
Fix according to spec + mrproper

JuRolland committed:
Fix according to spec + mrproper

JuRolland committed:
Valid test file

JuRolland committed:
Valid test file

yurug committed:
Cours 04/02.

yurug committed:
Cours 04/02.

yurug committed:
Defi 2.

yurug committed:
Defi 2.

n3f4s committed:
Split debug and from-scratch docker build

n3f4s committed:
Split debug and from-scratch docker build

n3f4s committed:
Remove trailing CLEAN var

n3f4s committed:
Remove trailing CLEAN var

JuRolland committed:
build_changes

JuRolland committed:
build_changes

n3f4s committed:
Put back deleted by mistake parameter Requiremetns

n3f4s committed:
Put back deleted by mistake parameter Requiremetns

JuRolland committed:
hotfix

JuRolland committed:
hotfix

yurug committed:
Version optimisée de generate.

yurug committed:
Makefile pour le generateur de grosses feuilles.

n3f4s committed:
Add Set for remembering already computed formula

n3f4s committed:
Add Set for remembering already computed formula

n3f4s committed:
Simplify time call for make perf

n3f4s committed:
Simplify time call for make perf

n3f4s committed:
Better make perf

n3f4s committed:
Better make perf

n3f4s committed:
Fix spelling in makefile

n3f4s committed:
Fix spelling in makefile

n3f4s committed:
Always build in release mode for perf test

n3f4s committed:
Add optional cell computation throughput benchmark

n3f4s committed:
Clean makefile and add variable to control bench

n3f4s committed:
Remove trailing set -x

JuRolland committed:
Partial implementation of Stylesheet

Commit:
premiers TU sur le graphe

Commit:
Code à trous pour Julien ;)

Commit:
premier TU

Commit:
TU élémentaires

Commit:
fin des tests élémentaires

Commit:
type optionnel pour l'évaluation

Commit:
test dependency_missing

JuRolland committed:
Not quite yet

JuRolland committed:
Crazy shit

n3f4s committed:
Fix missing mut

n3f4s committed:
Fix missing deref

n3f4s committed:
Not recursive function -> no borrow checking error

JuRolland committed:
Fix clone()

JuRolland committed:
spreadsheet v1

Commit:
Paramétrage de la taille de la ligne

n3f4s committed:
Fix derives

n3f4s committed:
We are not writing Java :)

Commit:
more tests

Commit:
test duplicated

JuRolland committed:
Spreadsheet is free

JuRolland committed:
Fix index

Commit:
printer où les changements ne sont pas encore implémentés

Commit:
Merge branch 'aquamen/master' of github.com:yurug/pcomp-2019 into aquamen/master

Commit:
print changes (beta)

Commit:
correction build_spreadsheet

JuRolland committed:
Fix bug

Commit:
coordonnées dans le bon sens

Commit:
coordonnées dans le bon sens

Commit:
intégration du printer

Commit:
intégration du printer

JuRolland committed:
Fix indexes

JuRolland committed:
Fix indexes

Commit:
Merge branch 'aquamen/master' of github.com:yurug/pcomp-2019 into aquamen/master

n3f4s committed:
Fix ordering of points

n3f4s committed:
Add first (not compiling) version of tree.rs

Commit:
dernieres corrections

Commit:
dernieres corrections

n3f4s committed:
Fix compilation error

n3f4s committed:
Finish Tree implementation

n3f4s committed:
remove comments

JuRolland committed:
Spreadsheet working

JuRolland committed:
Spreadsheet working

n3f4s committed:
Add ctor to tree

n3f4s committed:
Add set_cell function to tree

n3f4s committed:
Remove commented code

n3f4s committed:
Split zones by lines instead of by rectangle

n3f4s committed:
make Tree pub

n3f4s committed:
Make the area grow when needed to

Commit:
correction i/o

Commit:
correction i/o

Commit:
correction des tests du parser

Commit:
correction des tests du parser

Commit:
fix de 2 warnings

Commit:
fix de 2 warnings

JuRolland committed:
Suppression fichier interpreter

JuRolland committed:
Suppression fichier interpreter

JuRolland committed:
Changement signature apply_change

JuRolland committed:
Changement signature apply_change

Commit:
correction printer

Commit:
correction printer

Commit:
safer printer

Commit:
safer printer

Commit:
Allègement de la mémoire nécessaire

Commit:
Allègement de la mémoire nécessaire

n3f4s committed:
Make rectangle bound of tree public

n3f4s committed:
remove trailing whitespace

Commit:
résolution pb d'imports

Commit:
résolution pb d'imports

n3f4s committed:
Replace HashMap by Tree in Spreadsheet::inner

n3f4s committed:
Add interface for Spreadsheet

n3f4s committed:
Move files into own dir

n3f4s committed:
Update make to reflect file movement

n3f4s committed:
Build then run time on exec

Commit:
un seul open

Commit:
un seul open

n3f4s committed:
Add logs

n3f4s committed:
Refactor code

n3f4s committed:
Put tests and datas in their own directory

n3f4s committed:
Refactor, add ctor with size, change node naming

n3f4s committed:
Comment

n3f4s committed:
Move 2 challenge in defis/2

n3f4s committed:
Add function to get sender (to cell eval counter)

n3f4s committed:
Add function to get sender (to cell eval counter)

n3f4s committed:
Build tree with size

n3f4s committed:
Small node

n3f4s committed:
Fix makefile

n3f4s committed:
Move area mgmt & dump into own module

n3f4s committed:
Remove trailing duplicate function

n3f4s committed:
Move id/filename from content to node

n3f4s committed:
Split outside of Node construction

n3f4s committed:
Add size function

n3f4s committed:
Refactor to debug tree

n3f4s committed:
Actually free memory from vals

Commit:
architecture png

Commit:
màj archi

Commit:
typo

n3f4s committed:
Integrate tree in evaluator

n3f4s committed:
Remove commented code

n3f4s committed:
Merge branch 'master' into aquamen/space-opti

Commit:
premiers nettoyages

Commit:
documentation

Commit:
doc

Commit:
doc

n3f4s committed:
Clean code

n3f4s committed:
Remove println

n3f4s committed:
Log using trace!

n3f4s committed:
Enable log

n3f4s committed:
Fix return issue

n3f4s committed:
Fix CSV generator

Commit:
typo

JuRolland committed:
Quelques commentaires

n3f4s committed:
Add comments

n3f4s committed:
Merge branch 'aquamen/space-opti' into aquamen/master

n3f4s committed:
Fix merge issues

n3f4s committed:
Fix warnings

n3f4s committed:
Improve makefile

n3f4s committed:
Add tiny README.md

n3f4s committed:
Fix typo
